homeassistant/components/media_player/firetv.py

```diff
@@ -4,166 +4,142 @@
+                          're-establish the ADB connection in the next update')
+            returns = None
+            self.firetv.adb = None
+        finally:
+            self.adb_lock = False
 
-    @property
-    def current_app(self):
-        """Return the current app."""
-        try:
-            response = requests.get(
-                DEVICE_APPS_URL.format(
-                    self.proto, self.host, self.port, self.device_id, 'current'
-                    ), timeout=10).json()
-            _current_app = response.get('current_app')
-            if _current_app:
-                return _current_app.get('package')
-
-            return None
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve current app for %s", self.device_id)
-            return None
+        return returns
 
-    @property
-    def running_apps(self):
-        """Return a list of running apps."""
-        try:
-            response = requests.get(
-                DEVICE_APPS_URL.format(
-                    self.proto, self.host, self.port, self.device_id, 'running'
-                    ), timeout=10).json()
-            return response.get('running_apps')
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve running apps for %s", self.device_id)
-            return None
-
-    def action(self, action_id):
-        """Perform an action on the device."""
-        try:
-            requests.get(DEVICE_ACTION_URL.format(
-                self.proto, self.host, self.port, self.device_id, action_id
-                ), timeout=10)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Action request for %s was not accepted for device %s",
-                action_id, self.device_id)
-
-    def start_app(self, app_name):
-        """Start an app."""
-        try:
-            requests.get(DEVICE_APPS_URL.format(
-                self.proto, self.host, self.port, self.device_id,
-                app_name + '/start'), timeout=10)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not start %s on %s", app_name, self.device_id)
+    return _adb_wrapper
 
 
 class FireTVDevice(MediaPlayerDevice):
     """Representation of an Amazon Fire TV device on the network."""
 
-    def __init__(self, proto, host, port, device, name):
+    def __init__(self, host, name, adbkey, get_source, get_sources):
         """Initialize the FireTV device."""
-        self._firetv = FireTV(proto, host, port, device)
+        from firetv import FireTV  # pylint: disable=no-name-in-module
+        from adb.adb_protocol import (
+            InvalidCommandError, InvalidResponseError, InvalidChecksumError)
+
+        self._host = host
         self._name = name
-        self._state = STATE_UNKNOWN
-        self._running_apps = None
+        self._adbkey = adbkey
```

Comment:
This instance attribute isn't used.

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/17767/commits/5a24514d62f59bd161f394de3d6b90c4cd75bfd8